src/mono/wasm/debugger/BrowserDebugProxy/DevToolsHelper.cs

```diff
@@ -515,14 +515,22 @@ public ExecutionContext GetCurrentContext(SessionId sessionId)
                 ? context
                 : throw new KeyNotFoundException($"No execution context found for session {sessionId}");
 
-        public bool TryGetCurrentExecutionContextValue(SessionId id, out ExecutionContext executionContext)
+        public bool TryGetCurrentExecutionContextValue(SessionId id, out ExecutionContext executionContext, bool ignoreDestroyedContext = true)
         {
             executionContext = null;
             if (!contexts.TryGetValue(id, out ConcurrentBag<ExecutionContext> contextBag))
                 return false;
             if (contextBag.IsEmpty)
                 return false;
-            executionContext = contextBag.Where(context => context.Id == contextBag.Where(context => context.Destroyed == false).Max(context => context.Id)).FirstOrDefault();
+            IEnumerable<ExecutionContext> validContexts = null;
+            if (ignoreDestroyedContext)
+                validContexts = contextBag.Where(context => context.Destroyed == false);
+            else
+                validContexts = contextBag;
+            if (!validContexts.Any())
+                return false;
+            int maxId = validContexts.Max(context => context.Id);
+            executionContext = contextBag.Where(context => context.Id == maxId).FirstOrDefault();
```

Comment:
do/can we use `SingleOrDefault` here?
And instead of `.Where` you can use the predicate directly in `FirstOrDefault`.